Sources/MongoSwift/BSON/Document.swift

```diff
@@ -43,9 +53,15 @@ public struct Document: ExpressibleByDictionaryLiteral, ExpressibleByArrayLitera
         return self.makeIterator().values
     }
 
+    /// Returns a `[KeyValuePair]` containing the key-value pairs stored in this `Document`.
+    public var keyValuePairs: [KeyValuePair] {
```

Comment:
is this part of the protocol, or just something we are adding for the sake of it? I think Array(doc) may return the same thing

Reply:
I added this because getting a `[KeyValuePair]` was a useful thing for implementing the actual protocol related methods. I transferred it over here because `keys` and `values` already existed, so it seemed only natural to also have it rest alongside the two.

Reply:
re: our offline discussion on `Array(doc)`: After getting help from the xcode debugger I was able to confirm that the initializer for `Array` does indeed call the index-based subscript in order to populate its buffer. What this means is that we will have to rely on the `DocumentIterator` to get the `[KeyValuePair]`, otherwise we'll have a situation where doing `Array(doc)` leads us to call `subscript(index)` which calls `Array(doc)` and so on and so forth.

This should honestly be fine, since if we are going to eventually conform to `Collection`'s runtime guarantees, then we won't even need `keyValuePairs` anymore.

Reply:
Yup, we don't need it if we opt for `DocumentIterator.subsequence()`, so I've removed them.